File: 054_Quiz - overriding equals/src/hr/unipu/java/AMultipleChoiceQuestion.java
package hr.unipu.java;

// Nasljeđivanje – eliminacija ponavljanja
public class AMultipleChoiceQuestion extends AbstractQuestion {
    private String[] correctAnswers;    // atribut polje string

    // konstruktor
    public AMultipleChoiceQuestion(String text, String[] correctAnswers) {
        super(text);
        this.correctAnswers = correctAnswers;
    }

    // Ova metoda je zakomentirana jer je sada dio apstraktne klase
//    public String getText() {
//        return text;
//    }

    // konkrenta implementacija nasljeđene metode deklarirane kao apstraktna u apstraktnoj klasi
    public boolean isCorrectAnswer(String answer) {
        for (int i = 0; i < correctAnswers.length; i++) {
            if(correctAnswers[i].equalsIgnoreCase(answer))
                return true;    //ako nađe samo jedan true, vratit će true (=logičko ILI) i izaći će iz metode
        }
        return false;

    }

}

File: 054_Quiz - overriding equals/src/hr/unipu/kotlin/AMultipleChoiceQuestion.kt
package hr.unipu.kotlin

// Nasljeđivanje – eliminacija ponavljanja
class AMultipleChoiceQuestion// konstruktor
(text: String, private val correctAnswers: Array<String>    // atribut polje string
) : AbstractQuestion(text) {

    // Ova metoda je zakomentirana jer je sada dio apstraktne klase
    //    public String getText() {
    //        return text;
    //    }

    // konkrenta implementacija nasljeđene metode deklarirane kao apstraktna u apstraktnoj klasi
    override fun isCorrectAnswer(answer: String): Boolean {
        for (i in correctAnswers.indices) {
            if (correctAnswers[i].equals(answer, ignoreCase = true))
                return true    //ako nađe samo jedan true, vratit će true (=logičko ILI) i izaći će iz metode
        }
        return false

    }

}
